moveit_ros/planning_interface/moveit_cpp/src/moveit_cpp.cpp

```diff
@@ -0,0 +1,333 @@
+  {
+    std::string error = "Unable to configure planning scene monitor";
+    ROS_FATAL_STREAM_NAMED(LOGNAME, error);
+    throw std::runtime_error(error);
+  }
+
+  robot_model_ = planning_scene_monitor_->getRobotModel();
+  if (!robot_model_)
+  {
+    std::string error = "Unable to construct robot model. Please make sure all needed information is on the "
+                        "parameter server.";
+    ROS_FATAL_STREAM_NAMED(LOGNAME, error);
+    throw std::runtime_error(error);
+  }
+
+  bool load_planning_pipelines = true;
+  if (load_planning_pipelines && !loadPlanningPipelines(opt.planning_pipeline_options))
+  {
+    std::string error = "Failed to load planning pipelines from parameter server";
+    ROS_FATAL_STREAM_NAMED(LOGNAME, error);
+    throw std::runtime_error(error);
+  }
+
+  // TODO(henningkayser): configure trajectory execution manager
+  trajectory_execution_manager_.reset(new trajectory_execution_manager::TrajectoryExecutionManager(
+      robot_model_, planning_scene_monitor_->getStateMonitor()));
+
+  ROS_INFO_NAMED(LOGNAME, "MoveItCpp running");
+}
+
+MoveItCpp::MoveItCpp(MoveItCpp&& other)
+{
+  other.clearContents();
+}
+
+MoveItCpp::~MoveItCpp()
+{
+  ROS_INFO_NAMED(LOGNAME, "Deleting MoveItCpp");
+  clearContents();
+}
+
+MoveItCpp& MoveItCpp::operator=(MoveItCpp&& other)
+{
+  if (this != &other)
+  {
+    this->robot_description_ = other.robot_description_;
+    this->node_handle_ = other.node_handle_;
+    this->tf_buffer_ = other.tf_buffer_;
+    this->robot_model_ = other.robot_model_;
+    this->planning_scene_monitor_ = other.planning_scene_monitor_;
+    other.clearContents();
+  }
+
+  return *this;
+}
+
+bool MoveItCpp::loadPlanningSceneMonitor(const PlanningSceneMonitorOptions& opt)
+{
+  planning_scene_monitor_.reset(
+      new planning_scene_monitor::PlanningSceneMonitor(opt.robot_description, tf_buffer_, opt.name));
+  // Allows us to sycronize to Rviz and also publish collision objects to ourselves
+  ROS_DEBUG_STREAM_NAMED(LOGNAME, "Configuring Planning Scene Monitor");
+  if (planning_scene_monitor_->getPlanningScene())
+  {
+    // Start state and scene monitors
+    ROS_INFO_NAMED(LOGNAME, "Listening to '%s' for joint states", opt.joint_state_topic.c_str());
+    planning_scene_monitor_->startStateMonitor(opt.joint_state_topic, opt.attached_collision_object_topic);
+    planning_scene_monitor_->startPublishingPlanningScene(planning_scene_monitor::PlanningSceneMonitor::UPDATE_SCENE,
+                                                          opt.publish_planning_scene_topic);
+    planning_scene_monitor_->startSceneMonitor(opt.monitored_planning_scene_topic);
+  }
+  else
+  {
+    ROS_ERROR_STREAM_NAMED(LOGNAME, "Planning scene not configured");
+    return false;
+  }
+  // TODO(henningkayser): fix and remove lines below
+  ros::spinOnce();
+  ros::Duration(0.5).sleep();  // when at 0.1, i believe sometimes vjoint not properly loaded
```

Comment:
is this comment old from my boilerplate code? it looks familar to me. i think we should remove the sleep.

Reply:
ping: remove the sleep